src/utilities/flow.ts

```diff
@@ -0,0 +1,126 @@
+import { context, createContextKey } from "@opentelemetry/api";
+import { ReadableSpan, Tracer } from "@opentelemetry/sdk-trace-node";
+
+import { FlowKernelRequest } from "../api/types/FlowKernelRequest";
+import {
+    HUMANLOOP_FILE_TYPE_KEY,
+    HUMANLOOP_LOG_KEY,
+    HUMANLOOP_PARENT_SPAN_CTX_KEY,
+    HUMANLOOP_PATH_KEY,
+    HUMANLOOP_TRACE_FLOW_CTX_KEY,
+    NestedDict,
+    generateSpanId,
+    jsonifyIfNotString,
+    writeToOpenTelemetrySpan,
+} from "../otel";
+import { InputsMessagesCallableType } from "./types";
+
+export function flowUtilityFactory<I, M, O>(
+    opentelemetryTracer: Tracer,
+    func: InputsMessagesCallableType<I, M, O>,
+    path: string,
+    flowKernel?: FlowKernelRequest,
+): {
+    (inputs: I, messages: M): O extends Promise<infer R> ? Promise<R> : Promise<O>;
+    path: string;
+    version: FlowKernelRequest;
+} {
+    // @ts-ignore
```

Comment:
do add comments explaining what ts-ignores are doing.

(or if adding them during dev to make progress, go through the diff after to add comments elaborating on the ones that are left over)

Reply:
ack removed